mclient/crmtgs.1
.\"
.\"
.\"
.\"
.\"
.\"
.TH CRMTGS 1 "16 October 2009" "MIT SIPB"
.SH NAME
crmtgs \- Update a ~/.meetings file with the long name of each meeting
.SH SYNOPSIS
.B crmtgs
.SH DESCRIPTION
.I crmtgs
searches a ~/.meetings file for any meeting specifiers which do not
have their canonical "long name" specified in the file.  If any are
found, they are updated with the canonical long name for the meeting
in question.  
.I crmtgs 
accepts no arguments.

.SH FILES
~/.meetings	Data file listing meetings that you attend.

.SH AUTHORS
Bill Sommerfeld, MIT Student Information Processing Board

.SH "SEE ALSO"
.PP
discuss(1)








debian/changelog
debathena-discuss (10.0.12-0debathena1) unstable; urgency=low

  * Stop using the obsolete AC_TYPE_SIGNAL. (Trac: #347)
  * Add missing manpages. (Trac: #266)

 -- Evan Broder <broder@mit.edu>  Sun, 15 Nov 2009 17:35:19 -0500

debathena-discuss (10.0.11-0debathena1) unstable; urgency=low

  * Move the single-letter checks in trn_expr.y:yylex to the bottom of the
    function so that they get caught after commands that start with those
    letters. (Trac: #346)

 -- Evan Broder <broder@mit.edu>  Sat, 29 Aug 2009 17:20:43 -0400

debathena-discuss (10.0.10-0debathena2) unstable; urgency=low

  * Depend on Kerberos-enabled zephyr instead of build-depending.

 -- Evan Broder <broder@mit.edu>  Mon, 27 Jul 2009 14:38:10 -0400

debathena-discuss (10.0.10-0debathena1) unstable; urgency=low

  * Patch discuss server to work without krb4. Patch by Mitch Berger.

 -- Evan Broder <broder@mit.edu>  Thu, 23 Jul 2009 21:00:55 -0400

debathena-discuss (10.0.9-0debathena1) unstable; urgency=low

  * Fix local/dumb authentication needed for mail delivery and local
    client usage. Patch by Mitch Berger. (Trac: #274)
  * Always build without krb4.

 -- Evan Broder <broder@mit.edu>  Wed, 22 Jul 2009 22:58:40 -0400

debathena-discuss (10.0.8-0debathena1) unstable; urgency=low

  * Add one-character aliases to discuss. Patch by Ben Kaduk. (Trac: #238)
  * Squash some lintian warnings by switching from ${Source-Version} to
    ${source:Version} and ${binary:Version}
  * Use HAVE_KRB4 and HAVE_KRB5 instead of KERBEROS and KERBEROS5.
  * Assorted changes to make discuss work with any combination of krb4 and
    krb5 (including neither).
  * Always build our discuss with krb5.

 -- Evan Broder <broder@mit.edu>  Mon, 18 May 2009 23:51:52 -0400

debathena-discuss (10.0.7-0debathena3) unstable; urgency=low

  * Configure the discuss inetd service.

 -- Evan Broder <broder@mit.edu>  Thu, 05 Mar 2009 15:17:20 -0500

debathena-discuss (10.0.7-0debathena2) unstable; urgency=low

  * Change DEB_AUTO_UPDATE_AUTOCONF to 2.50, not 1.
  * Remove DEB_AUTO_UPDATE_DEBIAN_CONTROL.
  * Change "Debian-Athena Project" to "Debathena Project".
  * Remove debian/control from version control.
  * Clean up debian/copyright.

 -- Tim Abbott <tabbott@mit.edu>  Sat, 24 Jan 2009 14:31:41 -0500

debathena-discuss (10.0.7-0debathena1) unstable; urgency=low

  * Revert a small part of the OS X com_err patch from 10.0.6 in favor of
    changes to aclocal.m4
  * Remove hard-coded paths from discuss-mode, in favor of getting them
    from autoconf.
  * Remove debathena-machtype as a dependency, as it's no longer needed.

 -- Evan Broder <broder@mit.edu>  Fri, 09 Jan 2009 21:47:33 -0500

debathena-discuss (10.0.6-0debathena1) unstable; urgency=low

  [ Evan Broder ]
  * Add com_err support for delete and discuss on OS X.

  [ Anders Kaseorg ]
  * Depend emacs | emacsen instead of emacs22 | emacs21 | emacsen, so that
    the emacs metapackage is pulled in by default.

 -- Anders Kaseorg <andersk@mit.edu>  Thu, 08 Jan 2009 16:35:04 -0500

debathena-discuss (10.0.5-0debathena1) unstable; urgency=low

  * Fix some trivial typos.

 -- Anders Kaseorg <andersk@mit.edu>  Tue, 19 Aug 2008 16:35:07 -0400

debathena-discuss (10.0.4-0debathena1) unstable; urgency=low

  * Fix a crash bug in edsc on 64-bit platforms.
  * Fix date display in discuss on 64-bit platforms.

 -- Greg Hudson <ghudson@mit.edu>  Mon, 30 Jun 2008 12:50:56 -0400

debathena-discuss (10.0.3-0debathena1) unstable; urgency=low

  * Fix a crash bug in edsc (int/long mismatch in short_time).
  * Put edsc crash dumps in /tmp instead of /usr/tmp.

 -- Greg Hudson <ghudson@mit.edu>  Wed, 11 Jun 2008 19:39:38 -0400

debathena-discuss (10.0.2-0debathena1) unstable; urgency=low

  * Stop assuming that long is 32 bits.
  * Start assuming that int is 32 bits.

 -- Greg Hudson <ghudson@mit.edu>  Wed, 11 Jun 2008 16:29:35 -0400

debathena-discuss (10.0.1-0debathena1) unstable; urgency=low

  * Fix obsolete names for Kerberos library functions, which no longer
    worked on Mac OS X.  Patch from broder.

 -- Anders Kaseorg <andersk@mit.edu>  Sun, 06 Apr 2008 19:25:13 -0400

debathena-discuss (10.0.0-0debathena2) unstable; urgency=low

  * Actually use SERVERDIR, BINDIR, and INFODIR in config.h.sed.  (The
    Makefile already interpolates these variables.)

 -- Anders Kaseorg <andersk@mit.edu>  Sun, 06 Apr 2008 17:28:52 -0400

debathena-discuss (10.0.0-0debathena1) unstable; urgency=low

  * Merged quilt patches into mainline Athena tree

 -- Shreevatsa R <vatsa@mit.edu>  Sun,  6 Apr 2008 15:54:22 -0400

debathena-discuss (9.4.41-0debathena2) unstable; urgency=low

  * Add emacs22 as a default alternative to emacs21 | emacsen.

 -- Anders Kaseorg <andersk@mit.edu>  Mon, 29 Oct 2007 21:40:29 -0400

debathena-discuss (9.4.41-0debathena1) unstable; urgency=low

  * New upstream release supporting Kerberos 5.

 -- Anders Kaseorg <andersk@mit.edu>  Thu, 23 Aug 2007 14:28:28 -0400

debathena-discuss (9.4.30-0debathena3) unstable; urgency=low

  * Section: debathena/net.

 -- Anders Kaseorg <andersk@mit.edu>  Tue, 17 Jul 2007 06:33:41 -0400

debathena-discuss (9.4.30-0debathena2) unstable; urgency=low

  * Move edsc to debathena-discuss package.
  * Don't link libreadline, because libss is stupid and tries to do so at
    runtime.

 -- Anders Kaseorg <andersk@mit.edu>  Tue, 17 Jul 2007 03:37:48 -0400

debathena-discuss (9.4.30-0debathena1) unstable; urgency=low

  * Initial release.

 -- Anders Kaseorg <andersk@mit.edu>  Tue, 30 Jan 2007 22:57:54 -0500


mclient/crmtgs.c
/*
 *
 *	Copyright (C) 1988, 1989 by the Massachusetts Institute of Technology
 *    	Developed by the MIT Student Information Processing Board (SIPB).
 *    	For copying information, see the file mit-copyright.h in this release.
 *
 */
/*
 *	$Id: crmtgs.c,v 1.7 1999-04-12 16:47:06 ghudson Exp $
 *
 *	Fill out a .meetings file with the primary name of all the
 *	meetings in it.  This requires that the meeting be accessible
 *	at this time; however, this program may be run several times
 *	to get it right..
 *
 */

#ifndef lint
static char rcsid_crmtgs_c[] =
    "$Id: crmtgs.c,v 1.7 1999-04-12 16:47:06 ghudson Exp $";
#endif /* lint */

#include <stdio.h>
#include <discuss/discuss.h>

extern char *getenv ();

main(argc, argv)
	int argc;
	char **argv;
{
	int n_matches, code;
	register name_blk *nbp;
	name_blk *set;
	int i;
	mtg_info info;
	char **aliasv;
	int naliases;
	
#if defined(__APPLE__) && defined(__MACH__)
	add_error_table(&et_dsc_error_table);
#else
	initialize_dsc_error_table();
#endif

	dsc_expand_mtg_set(NULL, "*", &set, &n_matches, &code);

	if (code)
		punt_code("Can't expand meeting set", code);

	if (!n_matches)
		punt("No meetings?");

	for (nbp = set; nbp < set + n_matches; ++nbp) {
		dsc_get_mtg_info(nbp, &info, &code);
		if (code) {
			printf("%s: %s\n", nbp->aliases[0],
			       error_message(code));
			continue;
		}
		if (strcmp(nbp->aliases[0], info.long_name)) {
			printf("Adding long_name %s to %s\n",
			       info.long_name, nbp->aliases[0]);

			/*
			 * Beware of dragons and fenceposts in the
			 * following code.
			 */
			aliasv = nbp->aliases;
			while (*aliasv++) continue;
			naliases = aliasv - nbp->aliases;
			nbp->aliases = (char **) realloc(nbp->aliases,
				++naliases * sizeof(nbp->aliases));
			aliasv = nbp->aliases + naliases;
			while(naliases--) {
				*aliasv = aliasv[-1];
				--aliasv;
			}
			if (aliasv != nbp->aliases) abort();
			*aliasv = info.long_name;
		} else printf("Long name %s already present\n", 
			      info.long_name);
	}
	dsc_update_mtg_set(NULL, set, n_matches, &code);
	if (code)
		punt_code("update_mtg_set failed", code);
}

punt(string)
	char *string;
{
	puts(string);
	exit(1);
}

punt_code(string, code)
	char *string;
	int code;
{
	printf("%s: %s", string, error_message(code));
	exit(1);
} 

	
